### PR TITLE
comments , modal , and other utilities

### PR DESCRIPTION
**Pull request by sashaboi**
WORK IN PROGRESS. 

Made a modal for adding comments , and figured out the logic to maintain comments.

### CODE REVIEW

File: src/context/Post-context.js [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
         return action.payload;
+      case 'ADD_COMMENT':
+        console.log('add comment reached with this data :', action.payload);
+
+        console.log(
+          'allposts: ',
+          state,
+          'add comment to this post: ',
+          localComments
+        );
```
Comment (yashpatel1797) [inline src/context/Post-context.js:28]: remove console log.

**Review by yashpatel1797 — APPROVED** [review]
code looks clean
file structure is good
design looks consistent

File: src/components/Modal/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    console.log('make comment', inputComment);
+    const token = localStorage.getItem('token');
+    if (token === null) {
+      navigate('/auth/login');
+    }
+    const header = { headers: { authorization: token } };
+    axios
+      .post(
+        `/api/comments/add/${localComments._id}`,
+        { commentData: inputComment },
```
Comment (vikrantkode) [inline src/components/Modal/Modal.jsx:30]: try to use try catch

**Review by vikrantkode — APPROVED** [review]
Some suggestions left  have  a look on them  fir good coding structure
- good to go

**Pull request merged**